src/averageAges.js

```diff
@@ -37,7 +43,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const findWomen = (woman) => {
+    return (withChildren === undefined) ? (woman['sex'] === 'f') : (
+      (woman['sex'] === 'f')
+      && people.some((mother) => mother['mother'] === woman['name']));
+  };
+  const onlyCorrectWomen = people.filter(findWomen);
+  const totalYear = (sum, person) => {
+    return sum + (person['died'] - person['born']);
+  };
+  const womenAverage = onlyCorrectWomen.reduce(totalYear, 0);
+
+  return (womenAverage / onlyCorrectWomen.length);
```

Comment:
DRY - don't repeat yourself
Move repeatable code to separate function with meaningful variable naming and reuse it